hoodie-client/src/main/java/com/uber/hoodie/func/ParquetReaderWrappedIterator.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) 2018 Uber Technologies, Inc. (hoodie-dev-group@uber.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *          http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.uber.hoodie.func;
+
+import com.uber.hoodie.exception.HoodieIOException;
+import java.io.IOException;
+import java.util.Iterator;
+import org.apache.parquet.hadoop.ParquetReader;
+
+/**
+ * This class wraps a parquet reader and provides an iterator based api to
+ * read from a parquet file. This is used in {@link BufferedIterator}
+ */
+public class ParquetReaderWrappedIterator<T> implements Iterator<T> {
+
+  // Parquet reader for an existing parquet file
+  private final ParquetReader<T> parquetReader;
+  // Holds the next entry returned by the parquet reader
+  private T next;
+  // Holds the current entry returned by the parquet reader
+  private T prev;
```

Comment:
I think you can implement this using a single variable `next` ?

Reply:
should we add a test for this file.

Reply:
yeah, done.